server/services/interventionsService.test.ts

```diff
@@ -1675,21 +1675,21 @@ pactWith({ consumer: 'Interventions UI', provider: 'Interventions Service' }, pr
 
   describe('getActionPlanAppointments', () => {
     const actionPlanAppointments = [
-      {
+      actionPlanAppointmentFactory.build({
```

Comment:
I think we might need to be more explicit here, in a couple of places:

- The `state` of the interaction, which is currently `'an action plan with ID e5ed2f80-dfe2-4bf3-b5c4-d8d4486e963d exists and it has 3 scheduled appointments'`. This says nothing about whether the appointments have had their attendance / behaviour recorded yet, so we can't expect those fields to be `null` or `submitted` to be `false`. So if we do want to include all those fields in the expected response from the interaction, then we need to say something like `'and it has 3 scheduled appointments, none of which have had feedback submitted yet'`.
- The state in which we're asking the factory to build an appointment. I don't think we should make too many assumptions about the value created by `actionPlanAppointmentFactory.build()` — here we're assuming that it returns appointments that haven't had their feedback recorded yet, but I don't think we can make that assumption. Perhaps adding a `scheduled()` trait to the factory would make it clear that these appointments are scheduled but that feedback hasn't yet been given.

Reply:
Perhaps I'm overthinking this actually, do you have any thoughts?

Reply:
I think what confused me (and led me to open this PR) was that despite the way we've designed our `Appointment` models with nullable values, I wasn't quite sure from reading the contracts what came back from the API (i.e. that there were always `null` values on everything on the nested `sessionFeedback` object.

What confused me was that the expected response didn't fail the tests when these fields weren't present on the response object we'd written - I think using the factories does make this clearer, as it's clear we're returning a whole object, rather than the handwritten JSON response (even though we hide all the fields in that `build` method). I think that as long as it's clear that we're returning our whole internal representation of an `Appointment` here, this is an improvement.

I do think there's something to be said for adding a state for the attendance feedback test to specify that it `doesn't have feedback submitted yet` (which I'll add in) but for the initial `getActionPlanAppointments` I don't think there's that much of a need to add that state, as we're not really asserting anything on that feedback yet (and it can be called when session feedback _is_ present in the response)